File: ts/ISA/ISA/ArcTypes/IdentifierSetters.ts
import { checkValidCharacters } from "./Identifier.js";
import { ArcAssay } from "./ArcAssay.js";
import { ArcStudy } from "./ArcStudy.js";
import { ArcInvestigation } from "./ArcInvestigation.js";

export function setAssayIdentifier(newIdentifier: string, assay: ArcAssay): ArcAssay {
    checkValidCharacters(newIdentifier);
    assay.Identifier = newIdentifier;
    return assay;
}

export function setStudyIdentifier(newIdentifier: string, study: ArcStudy): ArcStudy {
    checkValidCharacters(newIdentifier);
    study.Identifier = newIdentifier;
    return study;
}

export function setInvestigationIdentifier(newIdentifier: string, investigation: ArcInvestigation): ArcInvestigation {
    checkValidCharacters(newIdentifier);
    investigation.Identifier = newIdentifier;
    return investigation;
}

File: src/ISA/ISA/ArcTypes/IdentifierSetters.fs
module ARCtrl.ISA.IdentifierSetters

open ARCtrl.ISA.Identifier

let setAssayIdentifier (newIdentifier: string) (assay: ArcAssay) =
    checkValidCharacters newIdentifier
    assay.Identifier <- newIdentifier
    assay

let setStudyIdentifier (newIdentifier: string) (study: ArcStudy) =
    checkValidCharacters newIdentifier
    study.Identifier <- newIdentifier
    study

let setInvestigationIdentifier (newIdentifier: string) (investigation: ArcInvestigation) =
    checkValidCharacters newIdentifier
    investigation.Identifier <- newIdentifier
    investigation
